app/models/book.rb

```diff
@@ -5,15 +5,17 @@ class Book < ActiveRecord::Base
   belongs_to :user
 
   validates :user, presence: true
+  validates_presence_of :title, :source_language, :target_language
```

Comment:
Let's add `null: false` to the columns here as well. These are useful for error messaging but don't give the same level of safety as a DB constraint.

Reply:
I didn't get this, When one of those attributes are missing an  "Your book is missing the following required details: [attributes]" error message is shown, If they are blank an "[attribute] can't be blank" error message is shown, So what do you mean by putting `null: false`.

Reply:
These validations only protect you at the application level, but they can't guarantee that bad data won't somehow slip into the database. Adding a null constraint at the DB level ensures that no null values ever get saved there. 

This line of code is good and we want to keep it, but we want to add a `change_column_null` in a migration alongside this.

https://apidock.com/rails/ActiveRecord/ConnectionAdapters/SchemaStatements/change_column_null